test/server/configuration_impl_test.cc

```diff
@@ -327,7 +357,10 @@ TEST(ConfigurationImplTest, NullTracerSetWhenTracingConfigurationAbsent) {
 
   NiceMock<Server::MockInstance> server;
   server.local_info_.cluster_name_ = "";
-  MainImpl config(server);
+  Upstream::ProdClusterManagerFactory cluster_manager_factory(
+      server.runtime(), server.stats(), server.threadLocal(), server.random(), server.dnsResolver(),
+      server.sslContextManager(), server.dispatcher(), server.localInfo());
+  MainImpl config(server, cluster_manager_factory);
```

Comment:
Why not support multiple constructors (including the old one), so can avoid the boilerplate here?